content/series/101/2_development_environment/index.md

```diff
@@ -0,0 +1,150 @@
+
+#### clang-format
+
+Most C++ programmers spend a considerable amount of time meticulously formatting their precious code, until they make an acquaintance with an autoformatter - of which `clang-format` is the indisputed champion - after which they pretty much never hand-format code again. This is the secret behind the reams of impossibly perfectly formatted and documented library code, for example throughout the standard library headers.
+
+It works by way of a configuration file in the root, much like `compile_commands.json`, called `.clang-format`, which the `clang-format` binary reads for the configured [**format style options**](https://clang.llvm.org/docs/ClangFormatStyleOptions.html). To start using it, create a `.clang-format` in the project root and select "Format document with clangd" through VSCode's command palette. VSCode also offers an option to auto-format documents on save (using the configured formatter).
+
+#### Compiler Warnings and clang-tidy
+
+Since C++ standards are generally very meticulous about backwards compatibility, not-so-great language and library design decisions made in the past cannot easily be rectified through breaking changes, and older programs must not suddenly stop compiling with a new compiler (provided they weren't engaging in unspecified behaviour in the first place). Ergo, there are a few language potholes and traps where the compiler can only warn you - cannot issue an error and stop (unless you configure it to be that way for a specific set of / all warnings), and which is why modern C++ development is greatly aided by turning up compiler warnings and using linters that perform static analysis outside compilation (perhaps even risking some false positive warnings). `clang++` offers a `g++`-like front-end, and accepts all those compiler flags (and many more); at the bare minimum it is highly recommended to set at least the following, for **every single C++ project you author**:
+
+```cmake
+target_compile_options(foo PUBLIC -Wall -Wextra)
+```
+
+`clang-tidy` can be invoked through the command line, though it is also available as a third-party [editor plugin](https://marketplace.visualstudio.com/items?itemName=notskm.clang-tidy) for VSCode (not authored by LLVM). There should be more integrated support for it within a few years.
+
+> _**Recommended:** [CppCon 2017: Kate Gregory "10 Core Guidelines You Need To Start Using Now"](https://www.youtube.com/watch?v=XkDEzfpdcSg)_
+
+### Building and Debugging
+
+The core idea behind debugging is to spin off a command line task: `cmake --build <output_dir>` which can be invoked by some keyboard shortcut via the text editor. Debugging is a bit more involved, though command line debugging is simpler on Linux than Windows; setting up VSCode to use the native debugger (`gdb` or `cppdbg`) is relatively straightforward on both. On VSCode both these aspects are driven by a JSON file (in `./.vscode`) each.
```

Comment:
> The core idea behind debugging is to spin off a command line task: `cmake --build <output_dir>` which can be invoked by some keyboard shortcut via the text editor.

Use different punctuation to guide the reading flow:

Better: _The core idea behind debugging is to spin off a command line task (`cmake --build <output_dir>`) which can be invoked by some keyboard shortcut via the text editor._